docs/ENVIRONMENT.md

```diff
@@ -129,6 +129,12 @@ All tools:
 * `$SYSTEMD_VERITY_SHARING=0` — if set, sharing dm-verity devices by
   using a stable `<ROOTHASH>-verity` device mapper name will be disabled.
 
+* `$SYSTEMD_OPENSSL_KEY_LOADER`— when using OpenSSL to load a key via an engine
+  or a provider, can be used to force the usage of one or the other interface.
+  Set to 'engine' to force the usage of the old engine API, and to 'provider'
+  force the usage of the new provider API. If unset, the provider will be tried
```

Comment:
maybe say explicitly that "provider API" is the way to go and the future, but that reality bites and hence "engine API" might have to be used instead. I.e. I think there#s value in explaining the situation briefly, since it's not obvious for the casual onlooker